src/api/image-builder.ts

```diff
@@ -79,24 +92,28 @@ export function getImages(): Promise<any> {
     fs.readdir(imagesDir).then(dirs => {
       docker.listImages()
         .then(images => {
-          const imgs = images.filter(image => {
-            if (image.RepoTags && image.RepoTags.length) {
-              const tag = image.RepoTags[0].split(':')[0];
-              return dirs.indexOf(tag) !== -1;
-            } else {
+          let imgs = dirs.map(d => {
+            let index = images.findIndex(i => {
+              if (i.RepoTags) {
+                return i.RepoTags.findIndex(t => t.startsWith(d)) !== -1;
+              }
+
               return false;
+            });
+            if (index !== -1) {
+              return {
+                name: d,
+                version: images[index].RepoTags.find(t => t.startsWith(d)).split(':')[1],
+                created: format(new Date(images[index].Created * 1000), 'DD.MM.YYYY HH:mm:ss'),
+                createdAgo: distanceInWordsToNow(new Date(images[index].Created * 1000)),
+                size: getHumanSize(images[index].Size),
+                dockerfile: null,
+                initsh: null
+              };
+            } else {
+              return null;
             }
-          }).map(image => {
-            return {
-              name: image.RepoTags[0].split(':')[0],
-              version: image.RepoTags[0].split(':')[1],
-              created: format(new Date(image.Created * 1000), 'DD.MM.YYYY HH:mm:ss'),
-              createdAgo: distanceInWordsToNow(new Date(image.Created * 1000)),
-              size: getHumanSize(image.Size),
-              dockerfile: null,
-              initsh: null
-            };
-          });
+          }).filter(d => d ? d : false);
```

Comment:
`.filter(Boolean);` is nicer and does the same thing. What do you think?

Reply:
did not know about that, awesome! :). I'll update that

Reply:
done